core/primitives.go

```diff
@@ -1033,3 +1034,25 @@ func (streams *PdfObjectStreams) WriteString() string {
 	b.WriteString(" 0 R")
 	return b.String()
 }
+
+func (d *PdfObjectDictionary) extractImage() (img *imageutil.ImageBase) {
```

Comment:
That's why I've made this a private method. At first I was expecting more encoders using image base, and this was used as a utility. 
Moved as a function with the input `PdfObjectDictionary` - it would be more natural now.